documentation/src/main/asciidoc/userguide/chapters/pc/PersistenceContext.adoc

```diff
@@ -857,4 +857,54 @@ include::{sourcedir}/CascadeOnDeleteTest.java[tags=pc-cascade-on-delete-example]
 ----
 include::{extrasdir}/pc-cascade-on-delete-example.sql[]
 ----
-====
\ No newline at end of file
+====
+
+[[pc-exception-handling]]
+=== Exception handling
+
+If the JPA `EntityManager` or the Hibernate-specific `Session` throws an exception, including any JDBC https://docs.oracle.com/javase/8/docs/api/java/sql/SQLException.html[`SQLException`], you have to immediately rollback the database transaction and close the current `EntityManager` or `Session`.
+
+Certain methods of the JPA `EntityManager` or the Hibernate `Session` will not leave the Persistence Context in a consistent state. As a rule of thumb, no exception thrown by Hibernate can be treated as recoverable. Ensure that the Session will be closed by calling the `close()` method in a finally block.
+
```

Comment:
I think the following text from the 4.2 documentation would be useful to add:

"Rolling back the database transaction does not put your business objects back into the state they were at the start of the transaction. This means that the database state and the business objects will be out of sync. Usually, this is not a problem because exceptions are not recoverable and you will have to start over after rollback anyway."

Reply:
I see that the above text is in "Session-per-application anti-pattern" section. Doesn't this apply in general?

Reply:
I added that 4.x paragraph as well. The above text is not in the `Session-per-application anti-pattern`, but in the `Exception handling` section.